lib/checks/aria/allowed-attr.js

```diff
@@ -6,7 +6,7 @@ var attr,
 	attrName,
 	allowed,
 	role = node.getAttribute('role'),
-	attrs = node.attributes;
+	attrs = axe.utils.getNodeAttributes(node);
```

Comment:
suggestion:
if we perhaps make `attributes` a getter in `virtualNode`, it does look seamless to access the property, same as what we have done for `isFocusable` or `tabbableElements`.

`node.attributes` can become `virtualNode.attributes`

Reply:
I think we should hold off on that for now since we're still figuring out what we want to put in the virtual node.